src/Microsoft.AspNetCore.Authentication/AuthenticationSchemeOptions.cs

```diff
@@ -36,5 +36,15 @@ public virtual void Validate()
         /// If set, will be used as the service type to get the Events instance instead of the property.
         /// </summary>
         public Type EventsType { get; set; }
+
+        /// <summary>
+        /// Used to ensure that the options are only initialized once.
+        /// </summary>
+        public bool Initialized { get; set; }
```

Comment:
I guess it would be clearer (and probably safer?) if options was responsible of the entire initialization process, 'cause that's likely the most appropriate place to determine whether options are singleton (= initialization should happen only once) or scoped (= initialization should happen for every instance).